src/test/java/net/snowflake/client/core/SnowflakeMFACacheTest.java

```diff
@@ -327,4 +333,28 @@ public void testUnavailableLocalSecureStorage() throws SQLException {
       CredentialManager.getInstance().resetSecureStorageManager();
     }
   }
+
+  // Run this test manually to test disabling the client request MFA token. Use an MFA
+  // authentication enabled user. This is valid for versions after 3.18.0.
+  @Test
+  @Ignore
+  public void testEnableClientRequestMfaToken() throws SQLException {
+    Map<String, String> params = AbstractDriverIT.getConnectionParameters();
+    SnowflakeBasicDataSource ds = new SnowflakeBasicDataSource();
+    ds.setServerName(params.get("host"));
+    ds.setAccount(params.get("account"));
+    ds.setPortNumber(Integer.parseInt(params.get("port")));
+    ds.setUser(params.get("user"));
+    ds.setPassword(params.get("password"));
+    ds.setEnableClientRequestMfaToken(false);
+
+    try (Connection con = ds.getConnection()) {
+      for (int i = 0; i < 3; i++) {
```

Comment:
we are setting the variables on login so I believe the `for` should be around connection creation not, only around executing statement

the same with the test for external browser